src/V3Class.cpp

```diff
@@ -181,8 +183,7 @@ class ClassVisitor final : public VNVisitor {
         // Give struct a pointer to its package and a final name
         dtypep->editCountInc();
         dtypep->classOrPackagep(m_classPackagep ? m_classPackagep : m_modp);
-        dtypep->name(dtypep->name() + (VN_IS(dtypep, UnionDType) ? "__union" : "__struct")
-                     + cvtToStr(dtypep->uniqueNum()));
+        dtypep->name(m_names.get(dtypep->name() + (VN_IS(dtypep, UnionDType) ? "__union" : "__struct")));
```

Comment:
Have you checked why the `dtypep` of these structs are the same? It seems to be wrong, because in the test, these structs have different sizes.

Or maybe there aren't the same, but their `uniqueNum` method returns the the same number. If so, I think we should change that method instead.

Generally, I think that the current approach (the one on master) should work, if `dtypep->uniqueNum()` returns a different number for different structs.

Reply:
It's right that they are not identical and do have the same uniqueNum. It's probably a bug in elaboration, if we find and fix it, we can use uniqueNum again. For now I think using UniqueNames.h this is good enough, because we excpect to visit every StructDType once anyway.

Reply:
If it is a problem only with `uniqueNum, I think we should change it instead. It probably may occur in other cases too.

Reply:
I think I see what is the problem. `m_uniqueNum` is declared in `AstNodeDType` and in some other types, which inherits from `AstNodeDType`. It is probably unnecessary, because one unique number for all types should be enough. The function that increments this number is defined only in `AstNodeDType`, so it increments only `m_uniqueNum`from that class. But `uniqueNum` method returns `uniqueNum` from the type on which it was called.